### PR TITLE
Add support for HBA drives in hpssa.chart.py

### PR DESCRIPTION
**Pull request by martinpal**
##### Summary

Add support for disks in HBA mode on HP controllers. 

##### Component Name

python.d/hpssa.chart.py

##### Test Plan

Manually tested in debug mode: /usr/libexec/netdata/plugins.d/python.d.plugin hpssa debug trace

Tested on HP ProLiant DL380p Gen8 with Smart Array P420i.

The following output proves that the values are parsed correctly

2020-10-17 21:11:21: python.d DEBUG: hpssa[hpssa] : Executing command '/usr/bin/sudo -n /usr/sbin/ssacli ctrl all show config detail'             
BEGIN hpssa.ctrl_temperature 4996794                                                                                                              
SET 'cache_0_temperature' = 41                                                                                                                    
SET 'ctrl_0_temperature' = 68                                                                                                                     
END                                                                                                                                               
                                                                                                                                                  
BEGIN hpssa.pd_status 4996794                                                                                                                     
SET 'pd_0_1I:1:2_status' = 1                                                                                                                      
SET 'pd_0_1I:1:12_status' = 1                                                                                                                     
SET 'pd_0_1I:1:7_status' = 1                                                                                                                      
SET 'pd_0_1I:1:3_status' = 1                                                                                                                      
SET 'pd_0_1I:1:9_status' = 1                                                                                                                      
SET 'pd_0_1I:1:10_status' = 1                                                                                                                     
SET 'pd_0_1I:1:5_status' = 1                                                                                                                      
SET 'pd_0_1I:1:1_status' = 1                                                                                                                      
SET 'pd_0_1I:1:8_status' = 1                                                                                                                      
SET 'pd_0_1I:1:4_status' = 1                                                                                                                      
SET 'pd_0_1I:1:11_status' = 1                                                                                                                     
SET 'pd_0_1I:1:6_status' = 1                                                                                                                      
END                                                                                                                                               
                                                                                                                                                  
BEGIN hpssa.pd_temperature 4996794                                                                                                                
SET 'pd_0_1I:1:6_temperature' = 25                                                                                                                
SET 'pd_0_1I:1:1_temperature' = 30                                                                                                                
SET 'pd_0_1I:1:12_temperature' = 27                                                                                                               
SET 'pd_0_1I:1:2_temperature' = 28                                                                                                                
SET 'pd_0_1I:1:10_temperature' = 29                                                                                                               
SET 'pd_0_1I:1:11_temperature' = 28                                                                                                               
SET 'pd_0_1I:1:5_temperature' = 27                                                                                                                
SET 'pd_0_1I:1:8_temperature' = 25                                                                                                                
SET 'pd_0_1I:1:3_temperature' = 27                                                                                                                
SET 'pd_0_1I:1:4_temperature' = 30                                                                                                                
SET 'pd_0_1I:1:9_temperature' = 23                                                                                                                
SET 'pd_0_1I:1:7_temperature' = 28                                                                                                                
END                                                                                                                                               
                                                                                                                                                  
BEGIN hpssa.ctrl_status 4996794                                                                                                                   
SET 'ctrl_0_status' = 1                                                                                                                           
SET 'battery_0_status' = 1                                                                                                                        
SET 'cache_0_status' = 0                                                                                                                          
END                                                                                                                                               
                                                                                                                                                  
BEGIN netdata.runtime_hpssa 4996794                                                                                                               
SET run_time = 1800                                                                                                                               
END                                                                                                                                               
                                                                                                                                                  
2020-10-17 21:11:23: python.d DEBUG: hpssa[hpssa] : update => [OK] (elapsed time: 1800, failed retries in a row: 0)                               


##### Additional Information

N/A

### CODE REVIEW

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- collectors/python.d.plugin/hpssa/hpssa.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=6356753&bid=20636891)

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- collectors/python.d.plugin/hpssa/hpssa.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=6356753&bid=20636891)

File: collectors/python.d.plugin/hpssa/hpssa.chart.py [hunk 1]
```diff
@@ -234,6 +236,15 @@ def parse_unassigned_physical_drives(self, adapter):
...
 
+    def parse_hba_physical_drives(self, adapter):
+        for line in self:
+            match = drive_regex.match(line)
+            if match:
+                self.parse_physical_drive(adapter, match.group('fqn'))
+            else:
+                self.rewind()
+                break
+
```
Comment (ilyam8) [inline collectors/python.d.plugin/hpssa/hpssa.chart.py:247]: i see this function equal to `parse_unassigned_physical_drives`

https://github.com/martinpal/netdata/blob/cc34080dcf10fcc0d74ea4cd81e53193646f383d/collectors/python.d.plugin/hpssa/hpssa.chart.py#L230-L246

How about to call it `parse_physical_drives` and use for both `line == 'Unassigned'` and `line == 'HBA Drives'`?

**Comment by ilyam8** [comment]
Hi @martinpal, thx for the contribution 👋

File: collectors/python.d.plugin/hpssa/hpssa.chart.py [hunk 2]
```diff
@@ -181,7 +181,7 @@ def parse_adapter(self, slot, adapter_type):
                 break
             elif array_regex.match(line):
                 self.parse_array(adapter)
-            elif line == 'Unassigned':
+            elif line == 'Unassigned' or line == 'HBA Drives':
                 self.parse_unassigned_physical_drives(adapter)
```
Comment (ilyam8) [inline collectors/python.d.plugin/hpssa/hpssa.chart.py:185]: lets rename `parse_unassigned_physical_drives` => `parse_physical_drives`

File: collectors/python.d.plugin/hpssa/hpssa.chart.py (hunk 1 above)
Comment (martinpal) [reply collectors/python.d.plugin/hpssa/hpssa.chart.py]: That would be better indeed. Updated. That makes my minor contribution even smaller - a one-line change.

File: collectors/python.d.plugin/hpssa/hpssa.chart.py (hunk 2 above)
Comment (ilyam8) [reply collectors/python.d.plugin/hpssa/hpssa.chart.py:185]: since it works not only for `unassigned`

**Review by ilyam8 — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
Thank you for you contribution @martinpal, lgtm!

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- collectors/python.d.plugin/hpssa/hpssa.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=6356753&bid=20636891)

**Pull request merged**